src/main/java/fi/aalto/cs/apluscourses/intellij/services/OpenSubmissionService.java

```diff
@@ -0,0 +1,127 @@
+package fi.aalto.cs.apluscourses.intellij.services;
+
+import static com.intellij.util.io.NettyKt.getOrigin;
+import static java.lang.Long.parseLong;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import fi.aalto.cs.apluscourses.intellij.utils.SubmissionDownloader;
+import fi.aalto.cs.apluscourses.utils.cache.CachePreferences;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.QueryStringDecoder;
+import java.io.IOException;
+import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.ide.RestService;
+
+public class OpenSubmissionService extends RestService {
```

Comment:
This service feels not very RESTful but I guess extending RestService is mostly for convenience.